### PR TITLE
[ML][Pipelines] fix: avoid raising OSError for invalid paths in LocalPathField deserialization

### PR DESCRIPTION
**Pull request by elliotzh**
# Description

For invalid path value like "azureml:/subscription/xxxxx", raise ValidationError instead of OSError, as OSError will directly break UnionField deserialization.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/tests/component/unittests/test_spark_component_entity.py [hunk 1]
```diff
@@ -103,3 +103,15 @@ def test_spark_component_version_as_a_function_with_inputs(self):
         rest_yaml_component = yaml_component._to_rest_object()
 
         assert rest_yaml_component == expected_rest_component
+
+    def test_from_rest_object(self):
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/component/unittests/test_spark_component_entity.py:107]: this is should be some common test? not just for spark?

File: sdk/ml/azure-ai-ml/tests/component/unittests/test_spark_component_entity.py [hunk 2]
```diff
@@ -103,3 +103,15 @@ def test_spark_component_version_as_a_function_with_inputs(self):
...
+        invalid_code = "azureml:/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.MachineLearningServices/" \
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/component/unittests/test_spark_component_entity.py:108]: why would we say this is invalid code? should be valid in rest result?

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/tests/component/unittests/test_spark_component_entity.py (hunk 2 above)
Comment (elliotzh) [reply sdk/ml/azure-ai-ml/tests/component/unittests/test_spark_component_entity.py:108]: you're right, let me update

File: sdk/ml/azure-ai-ml/tests/component/unittests/test_spark_component_entity.py (hunk 1 above)
Comment (elliotzh) [reply sdk/ml/azure-ai-ml/tests/component/unittests/test_spark_component_entity.py]: I have moved it to `test_component_schema.py` and test it with helloworld component (a command component)

**Pull request merged**